llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -898,11 +898,11 @@ class TargetTransformInfo {
 
   /// Estimate the overhead of scalarizing an instruction. Insert and Extract
   /// are set if the demanded result elements need to be inserted and/or
-  /// extracted from vectors.
-  InstructionCost getScalarizationOverhead(VectorType *Ty,
-                                           const APInt &DemandedElts,
-                                           bool Insert, bool Extract,
-                                           TTI::TargetCostKind CostKind) const;
+  /// extracted from vectors.  The involved values may be passed in VL if
+  /// Insert is true.
+  InstructionCost getScalarizationOverhead(
+      VectorType *Ty, const APInt &DemandedElts, bool Insert, bool Extract,
+      TTI::TargetCostKind CostKind, ArrayRef<Value *> VL = std::nullopt) const;
```

Comment:
```suggestion
      TTI::TargetCostKind CostKind, ArrayRef<Value *> VL = {}) const;
```
IIRC, std::nullopt is going to be deprecated/dleted for ArrayRef initialization

Reply:
fixed